### PR TITLE
Ability additional info is displayed and editable in front end

### PR DESCRIPTION
**Pull request by garunagiri**
This PR adds on to https://github.com/mitre/caldera/pull/1585

### CODE REVIEW

File: templates/profiles.html [hunk 1]
```diff
@@ -561,6 +611,20 @@ <h4 id="name" style="margin:5px; margin-top:5px;"></h4>
             keepOrder: true });
     }
 
+    function addAdditionalInfo(){
+        let $table_row = $('#additional-info-row-template-holder').find('#additional-info-template').clone();
+        $('#ability-additional-info').find('#additional-info-table').append($table_row);
+        $('#ability-additional-info').show();
+        $table_row.find('.ability-remove').click(function(){
+<!--    TODO: if deleting last row, hide additional info div    -->
```
Comment (privateducky) [inline templates/profiles.html:619]: no TODO's.. if we need to do it, let's do it

File: templates/profiles.html [hunk 2]
```diff
@@ -311,6 +337,18 @@ <h4 id="name" style="margin:5px; margin-top:5px;"></h4>
         data['tactic'] = $('#ability-tactic-name').val();
         data['technique'] = {'attack_id': $('#ability-tech-id').val(), 'name': $('#ability-tech-name').val()};
         data['platforms'] = platforms;
+
+        $('#ability-additional-info').find('#additional-info-table tr').each(function() {
+            let key = $(this).find('.additional-key').val();
+            if (key !== ''){
```
Comment (privateducky) [inline templates/profiles.html:343]: these two if checks can be combined

File: templates/profiles.html [hunk 3]
```diff
@@ -311,6 +337,18 @@ <h4 id="name" style="margin:5px; margin-top:5px;"></h4>
...
+                if (key in data){
+                    alert('Cannot save additional field! \'' + key + '\' in additional info is a reserved keyword.');
```
Comment (privateducky) [inline templates/profiles.html:345]: don't alert, use the alert modal that we use elsewhere

**Comment by privateducky** [comment]
the UI location doesn't feel quite right in this PR.

let's move it to the right, just under the 4 circles (add executor, upload payload, etc.) and give it a little more padding/space.

File: templates/profiles.html (hunk 1 above)
Comment (garunagiri) [reply templates/profiles.html:619]: Ugh... I actually did implement it, just forgot to delete the TODO.

File: templates/profiles.html (hunk 2 above)
Comment (garunagiri) [reply templates/profiles.html]: Combined now

File: templates/profiles.html (hunk 3 above)
Comment (garunagiri) [reply templates/profiles.html]: Using the duk-modal instead of alert. I hope that's the one you meant...

**Comment by garunagiri** [comment]
I've addressed the comments so far, should be good for another review

**Comment by privateducky** [comment]
looking better. a few visual tweaks:

1. remove the 'additional info' header along with the 'key' and 'value' header below it. instead, add a column to the left of the 'key' placeholder text called "info", similar to how we show named columns elsewhere on the page. 
2. move the "add info" button/image up a row, so there are 4 options in the row (generate new id, add executor, etc.). you may have to tweak these options to be smaller overall to fit it.

**Comment by garunagiri** [comment]
They actually were all on one row on my screen, but the spacings were hardcoded, which might be why they look differently on different screens. I will try to find a way to fix that, thanks!

**Comment by garunagiri** [comment]
I think I addressed tweak 1 correctly... And tweak 2 should be good. I checked with a smaller window

**Pull request merged**